src/test/java/uk/nhs/hee/tis/trainee/credentials/api/VerifyResourceTest.java

```diff
@@ -0,0 +1,206 @@
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import uk.nhs.hee.tis.trainee.credentials.SignatureTestUtil;
+import uk.nhs.hee.tis.trainee.credentials.dto.IdentityDataDto;
+import uk.nhs.hee.tis.trainee.credentials.filter.FilterConfiguration;
+import uk.nhs.hee.tis.trainee.credentials.service.VerificationService;
+
+@WebMvcTest(VerifyResource.class)
+@ComponentScan(basePackageClasses = FilterConfiguration.class)
+class VerifyResourceTest {
+
+  private static final String UNSIGNED_IDENTITY_DATA = """
+      {
+            "givenName": "Anthony",
+            "familyName": "Gilliam",
+            "birthDate": "1991-11-11",
+            "signature": {
+              "signedAt": "%s",
+              "validUntil": "%s"
+            }
+          }
+      """.formatted(Instant.MIN, Instant.MAX);
+
+  @Autowired
+  private MockMvc mockMvc;
+
+  @MockBean
+  private VerificationService service;
+
+  @MockBean
+  private RestTemplateBuilder restTemplateBuilder;
+
+  @Value("${application.signature.secret-key}")
+  private String secretKey;
+
+  @Test
+  void shouldForbidUnsignedVerifyIdentityRequests() throws Exception {
+    mockMvc.perform(
+            post("/api/verify/identity")
+                .content(UNSIGNED_IDENTITY_DATA)
+                .contentType(MediaType.APPLICATION_JSON))
+        .andExpect(status().isForbidden());
+  }
+
+  @Test
+  void shouldReturnCreatedWhenVerificationStarted()
+      throws Exception {
+    String signedData = SignatureTestUtil.signData(UNSIGNED_IDENTITY_DATA, secretKey);
+
+    String uriString = "the-uri";
+    URI uri = URI.create(uriString);
+    when(service.startIdentityVerification(any(IdentityDataDto.class), any())).thenReturn(uri);
+
+    mockMvc.perform(
+            post("/api/verify/identity")
+                .content(signedData)
+                .contentType(MediaType.APPLICATION_JSON))
+        .andExpect(status().isCreated())
+        .andExpect(header().string(HttpHeaders.LOCATION, uriString))
+        .andExpect(content().string(uriString));
+  }
+
+  @Test
+  void shouldPassStateDownstreamWhenStateGiven() throws Exception {
+    String signedData = SignatureTestUtil.signData(UNSIGNED_IDENTITY_DATA, secretKey);
+
+    ArgumentCaptor<String> stateCaptor = ArgumentCaptor.forClass(String.class);
+    when(service.startIdentityVerification(any(IdentityDataDto.class),
+        stateCaptor.capture())).thenReturn(URI.create(""));
+
+    mockMvc.perform(
+        post("/api/verify/identity")
+            .queryParam("state", "some-state-value")
+            .content(signedData)
+            .contentType(MediaType.APPLICATION_JSON));
+
+    String state = stateCaptor.getValue();
+    assertThat("Unexpected state.", state, is("some-state-value"));
```

Comment:
This is a bit of a random comment, but I don't have a very clear sense of what we'll be doing with `state`. 

In `VerificationServiceTest.shouldIncludeStateWhenStartingVerification()` we insist it should be a UUID, which conforms to the example we received originally from Sitekit which included `&state={{$guid}}` , but the swagger doc is a bit more vague ('`string`'). Although, that doc also says `string` for other things that seem they should sensibly be a guid, e.g. `nonce`, so maybe it is worth taking with a pinch of salt.

Reply:
The rough flow of requests is:
 1. The client calls `/api/verify/identity`, with an optional state `client-state`
 2. We forward the user to the gateway, with an internal state `internal-state`
 3. The callback is returned to `api/verify/callback`, including `internal-state`
 4. We get the token, perform verification and return the result to the client, with the previous `client_state`
 
We need to cache data against the state passed with request 2, that allows us to recover the same data when the callback is hit.
We can't simply pass through the original client state for security reasons, so we must generated our own state for the internal calls - which we can dictate is a UUID string.

The original `client-state` is also cached against the generated `internal-state`, which allows us to return it back to the original client at the end.

Reply:
Ok, that is clear, thanks. So client-state can be anything we want - any idea what we might use it for? I guess it allows us to supplement the user's session / app state context, I'm just not sure whether it's something we see an immediate use for, or whether it's just a way of passing state around that we might use in future if the need arises.

Reply:
It's dependant on what we're able to do on the frontend, but one immediate thought is being able to return the user back to the issuing dialog after verifying their identity.

If we assume this flow:
 1. User views their placements
 2. Selects to issue placement `xyz`
 3. A confirmation dialog is displayed for placement `xyz`
 4. They confirm that's what they want to issue
 5. We prompt for identity confirmed and route them in to that flow
 6. Backend/Gateway stuff happens
 7. They are returned to the app

At this point we have no way to identity where they were in the process, so we'd just send them back to the profile/landing page and expect them to go find their placement and start the process again.

However, we could include a state `client-state-xyz` at step 5 and store something like this in the local storage:
```json
{
  "client-state-xyz": {
    "path": "/issue/placement/xyz"
  }
}
```

Then at step 7 `client-state-xyz` is returned as a query param, we can extract it and use it to retrieve the local session data. Using that to restore the user's session to the same point they left.